scarb/src/core/registry/index/record.rs

```diff
@@ -15,6 +15,8 @@ pub struct IndexRecord {
     pub checksum: Checksum,
     #[serde(default = "default_false", skip_serializing_if = "is_false")]
     pub no_core: bool,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub yanked: Option<bool>,
```

Comment:
It should just deserialize to bool, since there is no difference between `None` and `Some(false)` (you can just copy the line from `no_core`).